bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/server/ICLanguageServerProvider.java

```diff
@@ -35,15 +38,14 @@ public interface ICLanguageServerProvider {
 	 * 
 	 * @param enableExpression
 	 */
-	public void setEnableExpression(EnableExpression enableExpression);
+	public void setEnableExpression(@Nullable EnableExpression enableExpression);
 	
 	/**
-	 * Check whether the LSP based C/C++ Editor and the language server shall be used for the given object.
-	 * Should be checked by the {@link EnableExpression#evaluate(Object)} method.
+	 * Check whether the LSP based C/C++ Editor and the language server shall be used for the given project.
 	 * 
-	 * @param receiver object which should be opened by the LSP Editor with language server
-	 * @return true if LSP based C/C++ Editor and language server shall be enabled for the given object
+	 * @param project 
+	 * @return true if LSP based C/C++ Editor and language server shall be enabled for the given project, otherwise false.
 	 */
-	public boolean isEnabledFor(Object receiver);
+	public boolean isEnabledFor(@Nullable IProject project);
```

Comment:
What is the gain to allow `null` here?

Reply:
A hint for the implementer that this is explicitly allowed to pass null.

Reply:
Yes, I understand what does it mean. I do not understand why do we allow 'null' argument. What is the benefit of asking if `ICLanguageServerProvider` is allowed for `null'?